docs/DeveloperGuide.md

```diff
@@ -5,6 +5,51 @@ title: Developer Guide
 * Table of Contents
 {:toc}
 
+## **Design**
+
+<div markdown="span" class="alert alert-primary">
+
+:bulb: **Tip:** The `.puml` files used to create diagrams in this document can be found in the [diagrams](https://github.com/se-edu/addressbook-level3/tree/master/docs/diagrams/) folder. Refer to the [_PlantUML Tutorial_ at se-edu/guides](https://se-education.org/guides/tutorials/plantUml.html) to learn how to create and edit diagrams.
+</div>
+
+### Architecture
+
+### Model component
+
+**API** : [`Model.java`](https://github.com/AY2223S2-CS2103T-T13-2/tp/blob/master/src/main/java/seedu/recipe/model/Model.java)
+
+<img src="images/ModelClassDiagram.png" width="500" />
+
+
+The `Model` component,
+
+* stores the recipe book data i.e., all `Recipe` objects (which are contained in a `UniqueRecipeBook` object).
+* stores the currently 'selected' `Recipe` objects (e.g., results of a search query such as `find` or `list`) as a 
+  separate _filtered_ list which is exposed to outsiders as an unmodifiable `ObservableList<Recipe>` instance that 
+  can be 'observed' e.g. the UI can be bound to this list so that the UI automatically updates when the data in the 
+  list changes.
+* stores a `UserPref` object that represents the user’s preferences. This is exposed to the outside as a 
+  `ReadOnlyUserPref` object.
+* does not depend on any of the other three components (as the `Model` represents data entities of the domain, they 
+  should make sense on their own without depending on other components)
+* However, it is worth noting that clients need only pass valid `IngredientBuilder` instances to `Recipe` objects, 
+  from which the population of the `Ingredient` and `IngredientQuantifier` fields will be done by the `Recipe` class
+  and its methods.
```

Comment:
The sentence from `from which` can be a bit harder to read. Can try ", where the `Recipe` class and its methods then populate the `Ingredient` and `IngredientQuantifier` fields".